### PR TITLE
Phase2-hgx88 Add a new method to provide layer #'s for each subdetector

### PR DESCRIPTION
**Pull request by bsunanda**
getLayer(const ForwardSubdetector type) returns # of layers for EE (type=HGCEE), FH (type=HGCHEF), BH (type=HGCHEB) and all combined (type=none of the 3 mentioned)

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
@bsunanda maybe it would be a good idea to introduce an enum for these types

**Comment by kpedro88** [comment]
or use the ForwardSubdetector enum https://github.com/cms-sw/cmssw/blob/master/DataFormats/ForwardDetId/interface/ForwardSubdetector.h

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
+1

**Comment by perrotta** [comment]
This is an extra welcome method, thank you Sunanda.

I'm wondering if it could be safer to reserve 0 (for example) to get the sum of all layers, and return -1 (for example) in case of any other input: just to be protected against misusage.

@bsunanda @kpedro88 : what do you think

**Comment by kpedro88** [comment]
@perrotta sounds good to me

**Comment by davidlange6** [comment]
please test

**Comment by bsunanda** [comment]
Since the method returns unsigned int, if the i/p parameter is not among the selected ones (HGCEE, HGCHEF, HGCHEB, ForwardEmpty) it returns a value 0

**Comment by kpedro88** [comment]
+1

File: RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc [hunk 1]
```diff
@@ -128,6 +128,29 @@ std::float_t RecHitTools::getRadiusToSide(const DetId& id) const {
   return size;
 }
 
+unsigned int RecHitTools::getLayer(const ForwardSubdetector type) const {
+
```
Comment (davidlange6) [inline RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc]: hi @bsunanda - please use a switch statement here.

**Comment by felicepantaleo** [comment]
thanks @bsunanda.

**Comment by davidlange6** [comment]
-1
pending answer to comment

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@perrotta @kpedro88 Could you approve this in view of latest changes as suggested by David

File: RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc [hunk 2]
```diff
@@ -130,24 +130,33 @@ std::float_t RecHitTools::getRadiusToSide(const DetId& id) const {
 
 unsigned int RecHitTools::getLayer(const ForwardSubdetector type) const {
 
-  int layer(0);
-  if        (type == ForwardSubdetector::HGCEE) {
+  int layer;
+  switch (type) {
+  case(ForwardSubdetector::HGCEE): {
```
Comment (perrotta) [inline RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc]: please indent the whole scope

File: RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc [hunk 3]
```diff
@@ -130,24 +130,33 @@ std::float_t RecHitTools::getRadiusToSide(const DetId& id) const {
...
     auto geomEE = static_cast<const HGCalGeometry*>(geom_->getSubdetectorGeometry(DetId::Forward,ForwardSubdetector::HGCEE));
     layer       = (geomEE->topology().dddConstants()).layers(true);
-  } else if (type == ForwardSubdetector::HGCHEF) {
+    break;
```
Comment (perrotta) [inline RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc]: why break? (here and below)

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc (hunk 3 above)
Comment (bsunanda) [inline RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc]: Removed them

File: RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc (hunk 2 above)
Comment (bsunanda) [inline RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc]: Done

**Comment by perrotta** [comment]
+1
Same as https://github.com/cms-sw/cmssw/pull/19709#issuecomment-315273473
Latest changes are fine with reco

File: RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc [hunk 4]
```diff
@@ -128,6 +128,35 @@ std::float_t RecHitTools::getRadiusToSide(const DetId& id) const {
...
 }
 
+unsigned int RecHitTools::getLayer(const ForwardSubdetector type) const {
+
+  int layer;
+  switch (type) {
+    case(ForwardSubdetector::HGCEE): {
+      auto geomEE = static_cast<const HGCalGeometry*>(geom_->getSubdetectorGeometry(DetId::Forward,ForwardSubdetector::HGCEE));
+      layer       = (geomEE->topology().dddConstants()).layers(true);
+    }
```
Comment (davidlange6) [inline RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc]: you need to add a break as part of each case. @bsunanda

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by perrotta** [comment]
+1
Same as https://github.com/cms-sw/cmssw/pull/19709#issuecomment-315273473
Latest changes are fine with reco

**Comment by kpedro88** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**